backend/de.metas.business/src/main/java/de/metas/product/IProductDAO.java

```diff
@@ -173,4 +173,5 @@ private ProductQuery(
 
 	int getProductGuaranteeDaysMinFallbackProductCategory(@NonNull final ProductId productId);
 
+	int getGuaranteeMonthsInDays(I_M_Product product);
```

Comment:
avoid using data records like I_M_Product, I_C_Order etc outside DAOs and Repositories. Instead, please use (in this case) ProductId as parameter